index.js

```diff
@@ -38,11 +38,11 @@ module.exports = function retryAsPromised(callback, options) {
     name: options.name || callback.name || 'unknown'
   };
 
-  var report = options.report;
-
-  // Massage match option into array so we can blindly treat it as such later
-  if (!Array.isArray(options.match)) options.match = [options.match];
+  if (typeof options.match !== 'function' && !Array.isArray(options.match)) {
```

Comment:
Support `[function]` just for conviences sake.

Reply:
@mickhansen Makes sense, will do